api/v1beta1/azuremanagedcontrolplane_types.go

```diff
@@ -186,6 +186,23 @@ type AzureManagedControlPlaneSpec struct {
 	// For authentication with Azure Container Registry.
 	// +optional
 	KubeletUserAssignedIdentity string `json:"kubeletUserAssignedIdentity,omitempty"`
+
+	// HTTPProxyConfig is the HTTP proxy configuration for the cluster.
+	// Immutable.
+	// +optional
+	HTTPProxyConfig *HTTPProxyConfig `json:"httpProxyConfig,omitempty"`
+}
+
+// HTTPProxyConfig is the HTTP proxy configuration for the cluster.
+type HTTPProxyConfig struct {
+	// HTTPProxy - The HTTP proxy server endpoint to use.
+	HTTPProxy *string `json:"httpProxy,omitempty"`
+	// HTTPSProxy - The HTTPS proxy server endpoint to use.
+	HTTPSProxy *string `json:"httpsProxy,omitempty"`
+	// NoProxy - The endpoints that should not go through proxy.
+	NoProxy *[]string `json:"noProxy,omitempty"`
```

Comment:
I think we can change this to a `[]string` instead a slice pointer since an empty slice would indicate it's not set. That way, we could just do `len(NoProxy) > 0` instead of `NoProxy != nil && len(NoProxy) > 0'

Reply:
https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/3847/files#r1296341246

Reply:
Is there anyway we can change this type though? I searched through the CAPZ repo and we don't really have any mentions of a pointer to a slice in any of the types in CAPZ.

Reply:
Yes, makes sense. This is one such instance where we make the conversion
```
	if s.APIServerAccessProfile != nil {
		managedCluster.APIServerAccessProfile = &containerservice.ManagedClusterAPIServerAccessProfile{
			EnablePrivateCluster:           s.APIServerAccessProfile.EnablePrivateCluster,
			PrivateDNSZone:                 s.APIServerAccessProfile.PrivateDNSZone,
			EnablePrivateClusterPublicFQDN: s.APIServerAccessProfile.EnablePrivateClusterPublicFQDN,
		}

		if s.APIServerAccessProfile.AuthorizedIPRanges != nil {
			managedCluster.APIServerAccessProfile.AuthorizedIPRanges = &s.APIServerAccessProfile.AuthorizedIPRanges
		}
	}
```
Let me make that change.